### PR TITLE
doc(talos): v2 engine support

### PR DESCRIPTION
**Pull request by c3y1huang**
#### Which issue(s) this PR fixes:

Issue longhorn/longhorn#7791

#### What this PR does / why we need it:

Add important notes and instructions outlining the v2 engine support for Talos Linux.

#### Special notes for your reviewer:

`None`

#### Additional documentation or context

`None`

### CODE REVIEW

**Review by derekbit — DISMISSED** [review]
LGTM

**Comment by derekbit** [comment]
Waiting for @jillian-maroket's review.

File: content/docs/1.8.0/advanced-resources/os-distro-specific/talos-linux-support.md [hunk 1]
```diff
@@ -56,9 +56,22 @@ machine:
 
 For detailed instructions, see the Talos documentation on [Editing Machine Configuration](https://www.talos.dev/v1.6/talos-guides/configuration/editing-machine-configuration/).
 
-## Limitations
+## v2 Data Engine
```
Comment (jillian-maroket) [inline content/docs/1.8.0/advanced-resources/os-distro-specific/talos-linux-support.md:59]:
```suggestion
## V2 Data Engine
```

File: content/docs/1.8.0/advanced-resources/os-distro-specific/talos-linux-support.md [hunk 2]
```diff
@@ -56,9 +56,22 @@ machine:
...
 
-- Exclusive to v1 data volume: currently, within a Talos Linux cluster, Longhorn only supports v1 data volume. The v2 data volume isn't currently supported in this environment.
+To use the v2 data engine volumes, ensure your nodes fulfill the v2 data engine [prerequisites](../../../v2-data-engine/prerequisites#prerequisites).
```
Comment (jillian-maroket) [inline content/docs/1.8.0/advanced-resources/os-distro-specific/talos-linux-support.md:61]:
```suggestion
To use V2 volumes, all nodes must meet the V2 Data Engine [prerequisites](../../../v2-data-engine/prerequisites#prerequisites).
```

File: content/docs/1.8.0/advanced-resources/os-distro-specific/talos-linux-support.md [hunk 3]
```diff
@@ -56,9 +56,22 @@ machine:
...
+  sysctls:
+    vm.nr_hugepages: "1024"
+  kernel:
+    modules:
+      - name: nvme_tcp
+      - name: vfio_pci
+#     - name: uio_pci_generic
+```
+
+**Note:** Talos Linux version 1.7.x and prior versions do not include the `uio_pci_generic` kernel module. If your system device supports `vfio_pci`, it's the preferred method and you won't need to install and enable the `uio_pci_generic` kernel driver, for more details, see [SPDK: System Configuration User Guide](https://spdk.io/doc/system_configuration.html). If `vfio_pci` is incompatible with your system or specific hardware, you will need to use `uio_pci_generic`. Future version of Talos Linux are expected to include native support for `uio_pci_generic`, see GitHub issue - [uio_pci_generic can be required for DPDK or SPDK](https://github.com/siderolabs/talos/issues/9236).
```
Comment (jillian-maroket) [inline content/docs/1.8.0/advanced-resources/os-distro-specific/talos-linux-support.md:74]:
```suggestion
> **Note:** 
> Talos Linux v1.7.x and earlier versions do not include the `uio_pci_generic` kernel module. If your system device supports `vfio_pci`, which is the preferred kernel module for SPDK application deployment, you are not required to install and enable the `uio_pci_generic` kernel driver. For more information, see [System Configuration User Guide](https://spdk.io/doc/system_configuration.html) in the SPDK documentation.
>
> You can use `uio_pci_generic` if `vfio_pci` is incompatible with your system or specific hardware. Future versions of Talos Linux are expected to include native support for `uio_pci_generic`. For more information, see [Issue #9236](https://github.com/siderolabs/talos/issues/9236).
```

File: content/docs/1.8.0/important-notes/_index.md [hunk 4]
```diff
@@ -193,6 +195,12 @@ To apply the storage network to existing RWX volumes, you must detach the volume
 
 For more information, see [Issue #8184](https://github.com/longhorn/longhorn/issues/8184).
 
+## OS Distro Specific
+
+### Talos Linux
+
+The v2 volumes are supported from Longhorn v1.8.0. To use the v2 data engine, setup your Talos Linux cluster to meet the v2 data engine prerequisites. For more information, see [Talos Linux Support: v2 data engine](../advanced-resources/os-distro-specific/talos-linux-support#v2-data-engine).
```
Comment (jillian-maroket) [inline content/docs/1.8.0/important-notes/_index.md:202]:
```suggestion
Longhorn v1.8.0 and later versions support usage of V2 volumes in Talos Linux clusters. To use V2 volumes, ensure that all nodes meet the V2 Data Engine prerequisites. For more information, see [Talos Linux Support: V2 Data Engine](../advanced-resources/os-distro-specific/talos-linux-support#v2-data-engine).
```

**Review by jillian-maroket — COMMENTED** [review]
Let me know if you have concerns about the changes.

**Comment by c3y1huang** [comment]
> Let me know if you have concerns about the changes.

The suggestions LGTM, I've made the changes. PTAL, thank you!

**Review by jillian-maroket — APPROVED** [review]
LGTM. Thanks!

**Review by innobead — APPROVED** [review]

**Pull request merged**